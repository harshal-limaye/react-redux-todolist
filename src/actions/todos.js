export const addToDo = text => ({ type: 'ADD', text });

export const completeToDO = id => ({ type: 'COMPLETE', id });

export const deleteToDO = id => ({ type: 'DELETE', id });