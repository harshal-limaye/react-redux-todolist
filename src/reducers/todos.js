const todos = (state = [], action) => {
    switch (action.type) {
        case 'ADD':
            return [ { id: state.length, text: action.text, completed: false }, ...state ];
        case 'COMPLETE':
            return state.filter(t => t.id === action.id ? Object.assign({}, t, {completed: !t.completed}) : t);
        case 'DELETE':
            return state.filter(t => t.id !== action.id);
        default:
            return state;
    }
}

export default todos;