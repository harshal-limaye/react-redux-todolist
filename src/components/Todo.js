import React from 'react';
import { connect } from 'react-redux';
import { completeToDO, deleteToDO } from '../actions/todos';

class Todo extends React.Component {
    render() {
        return (
            <li className={this.props.todo.completed ? 'complete' : ''}>
                <input 
                    onClick={this.onComplete} 
                    defaultChecked={this.props.todo.completed} 
                    type="checkbox" />
                {this.props.todo.text}
                <button 
                    onClick={this.onDelete} 
                    className="btn btn-delete"
                >Delete</button>
            </li>
        );
    }

    onComplete = id => this.props.onComplete(this.props.todo.id);

    onDelete = id => this.props.onDelete(this.props.todo.id);
}

export default Todo;