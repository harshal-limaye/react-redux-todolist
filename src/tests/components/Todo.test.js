import React from 'react';
import { shallow } from 'enzyme';
import Todo from '../../components/Todo';
import todos from '../fixtures/todos';

let wrapper;

describe('testing todo component', () => {
    beforeEach(() => {
        wrapper = shallow(<Todo todo={todos[0]} />)
    });
    
    test('should render todo component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should check if active class name applied', () => {
        expect(wrapper.find('.complete').length).toBe(1);
    });

    test('should not apply class name `complete` if completed is false', () => {
        const component = shallow(<Todo todo={todos[3]} />);
        expect(component.find('.complete').length).toBe(0);
    });

    test('should call onDelete method', () => {
        const onDelete = jest.fn();
        const component = shallow(<Todo todo={todos[3]} onDelete={onDelete} />);
        component.instance().onDelete(todos[3].id);
        expect(onDelete).toHaveBeenCalled();
    });

    test('should call onComplete method', () => {
        const onComplete = jest.fn();
        const component = shallow(<Todo todo={todos[3]} onComplete={onComplete} />);
        component.instance().onComplete(todos[3].id);
        expect(onComplete).toHaveBeenCalled();
    });
});