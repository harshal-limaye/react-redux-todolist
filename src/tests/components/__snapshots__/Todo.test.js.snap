// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing todo component should render todo component correctly 1`] = `
<li
  className="complete"
>
  <input
    defaultChecked={true}
    onClick={[Function]}
    type="checkbox"
  />
  Write test cases
  <button
    className="btn btn-delete"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
